### PR TITLE
refactor(core): Convert executions to react

### PR DESCRIPTION
**Pull request by jrsquared**

### CODE REVIEW

File: app/scripts/modules/core/src/delivery/delivery.states.ts [hunk 1]
```diff
@@ -67,7 +67,7 @@ module(DELIVERY_STATES, [
     url: `?${stateConfigProvider.paramsToQuery(filterModelConfig)}`,
     views: {
       'pipelines': {
-        template: '<executions application="$resolve.app"></executions>',
+        component: Executions, $type: 'react',
```
Comment (anotherchrisberry) [inline app/scripts/modules/core/src/delivery/delivery.states.ts]: separate lines here, or just inline the whole thing, i.e. `{ component: Executions, $type: 'react' }`?

File: app/scripts/modules/core/src/delivery/executions/Executions.tsx [hunk 2]
```diff
@@ -0,0 +1,373 @@
...
+    super(props);
+
+    const { app } = props;
+    const { executionFilterModel, insightFilterStateModel } = ReactInjector;
+    if (executionFilterModel.mostRecentApplication !== app.name) {
+      executionFilterModel.asFilterModel.groups = [];
+      executionFilterModel.mostRecentApplication = app.name;
+    }
+
+    if (props.app.notFound) { return; }
```
Comment (anotherchrisberry) [inline app/scripts/modules/core/src/delivery/executions/Executions.tsx]: just `(app.notFound)`

File: app/scripts/modules/core/src/delivery/executions/Executions.tsx [hunk 3]
```diff
@@ -0,0 +1,373 @@
...
+        const monitor = ReactInjector.executionService.waitUntilNewTriggeredPipelineAppears(this.props.app, newPipelineId);
+        monitor.then(() => this.setState({ triggeringExecution: false }));
+      },
+      () => this.setState({ triggeringExecution: false })
+    );
+  };
+
+  private triggerPipeline(): void {
+    ReactGA.event({category: 'Pipelines', action: 'Trigger Pipeline (top level)'});
+    // TODO: Convert the modal to react
```
Comment (anotherchrisberry) [inline app/scripts/modules/core/src/delivery/executions/Executions.tsx]: // TODO: Have fun with that.

Comment (jrsquared) [reply app/scripts/modules/core/src/delivery/executions/Executions.tsx]: baby steps.

**Comment by anotherchrisberry** [comment]
LGTM

**Pull request merged**